File: TabletopStats.Infrastructure/Entity/SessionLog.cs
using System.ComponentModel.DataAnnotations;

namespace TabletopStats.Infrastructure.Entity;

public class SessionLog
{
    [Key]
    public Guid SessionId { get; set; }
    
    [Required]
    public string SessionName { get; set; }
    
    [Required]
    public DateTime StartTime { get; set; }
    
    [Required]
    public DateTime EndTime { get; set; }
    
    [Required]
    public RpgSystem RpgSystem { get; set; }
    
    // public Guid? GameMasterId { get; set; }
    public Person? GameMaster { get; set; }
    
    [Required]
    public ICollection<Person> Players { get; set; }
}

File: TabletopStats.UI.React/src/models/SessionLog.ts
import { Row } from "simple-table-core";
import { Player } from "./Player";
import { RpgSystem } from "./RpgSystem";
import { GUID } from "./types/Guid";
import { TimeSpan } from "./types/timepan";

export interface SessionLog {
    sessionId: GUID,
    adventureId?: GUID,
    campaignId?: GUID,
    sessionName: string,
    description: string,
    startTime: string,
    endTime: string,
    duration: string,
    players: Player[],
    gameMaster?: Player,
    rpgSystem: RpgSystem
}

export interface SessionLogParsed extends Row {
    sessionId: GUID,
    sessionName: string,
    description: string,
    parsedStartTime: string,
    parsedEndTime: string,
    duration: string,
    playerNames: string[],
    gameMasterName?: string,
    rpgSystemName: string
}